protobuf.lua

```diff
@@ -0,0 +1,692 @@
+    prjDir .. "/src/google/protobuf/stubs/statusor.h",
+    prjDir .. "/src/google/protobuf/stubs/stringpiece.h",
+    prjDir .. "/src/google/protobuf/stubs/stringprintf.h",
+    prjDir .. "/src/google/protobuf/stubs/strutil.h",
+    prjDir .. "/src/google/protobuf/stubs/time.h",
+    prjDir .. "/src/google/protobuf/wire_format_lite.h",
+  }
+
+  includedirs {
+    prjDir .. "/../zlib",
+    prjDir .. "/src"
+  }
+
+  -- -------------------------------------------------------------
+  -- configurations
+  -- -------------------------------------------------------------
+
+  if (os.is("windows") and not _TARGET_IS_WINRT and not _TARGET_IS_WINPHONE) then
+    -- -------------------------------------------------------------
+    -- configuration { "windows" }
+    -- -------------------------------------------------------------
+
+    -- common configuration settings
+
+    dofile (_BUILD_DIR .. "/static_win.lua")
+
+    -- project specific configuration settings
+
+    -- configuration { "windows" }
+
+    -- -------------------------------------------------------------
+    -- configuration { "windows", "Debug", "x32" }
+    -- -------------------------------------------------------------
+
+    -- common configuration settings
+
+    dofile (_BUILD_DIR .. "/static_win_x86_debug.lua")
+
+    -- project specific configuration settings
+
+    -- configuration { "windows", "Debug", "x32" }
+
+    -- -------------------------------------------------------------
+    -- configuration { "windows", "Debug", "x64" }
+    -- -------------------------------------------------------------
+
+    -- common configuration settings
+
+    dofile (_BUILD_DIR .. "/static_win_x64_debug.lua")
+
+    -- project specific configuration settings
+
+    -- configuration { "windows", "Debug", "x64" }
+
+    -- -------------------------------------------------------------
+    -- configuration { "windows", "Release", "x32" }
+    -- -------------------------------------------------------------
+
+    -- common configuration settings
+
+    dofile (_BUILD_DIR .. "/static_win_x86_release.lua")
+
+    -- project specific configuration settings
+
+    -- configuration { "windows", "Release", "x32" }
+
+    -- -------------------------------------------------------------
+    -- configuration { "windows", "Release", "x64" }
+    -- -------------------------------------------------------------
+
+    -- common configuration settings
+
+    dofile (_BUILD_DIR .. "/static_win_x64_release.lua")
+
+    -- project specific configuration settings
+
+    -- configuration { "windows", "Release", "x64" }
+
+    -- -------------------------------------------------------------
+  end
+
+  if (os.is("linux") and not _OS_IS_ANDROID) then
+    -- -------------------------------------------------------------
+    -- configuration { "linux" }
+    -- -------------------------------------------------------------
+
+    -- common configuration settings
+
+    defines { "HAVE_PTHREAD" }
```

Comment:
I've had a few problems with the new template. "shared_project.lua" tries to add all source files in a project which we want to avoid with protobuf. "3rdparty_<static/shared>_project.lua" doesn't which is why I went with it in the beginning. Will all 3rdParty libraries have to use "shared_project.lua" from now on? Also I've noticed that we are including the shared versions of base lua files for the various different platforms, except for WinUWP, but we are including a the base lua script for shared projects. Is this by design? I thought we are going with static linking by default for 3rd party libs although I've noticed in the output directory there are a handful of dynamic ones. The last thing I noticed is that the iOS config settings is missing.

One last thing is that this is giving me errors on windows builds where the jenkins slave expects cygwin installed. I'm still working through it and I've most likely done something wrong, but I thought I'd mention it.

Reply:
@mfpdias I should probably create a 3rdparty template as well to disambiguate these.  You are correct in that we should be using the 3rdparty_static_project.lua instead of the main one.  I was referrering to the template in terms of which architectures to drop and which ones to add so sorry about the confusion that I may have caused.  The file is almost exactly where I think it should be aside from my previous comment
```
    dofile (_BUILD_DIR .. "/static_linux.lua")

     -- project specific configuration settings

     configuration { "linux" }

      defines { "HAVE_PTHREAD" }
```

Reply:
Sorry, no that was my fault. I've removed the non architecture related changes and have been incrementally adding the architecture changes to the old lua premake script from the one you provided and I think it's almost at a point where it will build. The 2 remaining things now are:
- The [strange missing Cygwin error](http://runtime-test.esri.com:8080/job/dev_3rdparty_vtest_win/48/console) I'm getting on the windows build slave which still happens with the old premake lua script and is making me think it's a windows build slave issue.
- C4996 deprecated posix function on WinUWP, which should just be a warning, so it's possible an '/sdl' flag might have snuck in there from a base lua config file, although I can't see where that might be happening.